### ISSUE TITLE
Scala files in the root project get compiled

### ISSUE THREAD

**Issue by japgolly** [scalacenter/bloop#598]
Using Bloop 1.0.0,

```sh
mkdir code
cd code

echo 'name := "blah"

scalaVersion := "2.12.6"

scalacOptions := List(
        "-unchecked",
        "-deprecation",
        "-Ypartial-unification",
        "-feature", "-language:postfixOps", "-language:implicitConversions", "-language:higherKinds", "-language:existentials")

triggeredMessage := Watched.clearWhenTriggered
' > build.sbt

mkdir project
echo 'sbt.version=1.1.6' > project/build.properties

echo 'object X{}' > x.scala

sbt bloopInstall

bloop compile -w code
```

then make changes to `x.scala`. It should trigger recompiles but doesn't.

Workaround is to move `x.scala` into `src/main/scala`

**Comment by jvican** [comment]
The mystery here is not that it doesn't recompile on file change (this is expected, as Bloop only listens to source directories), but the mere fact that both sbt and bloop compile Scala files in the root project! This is not expected: only Scala files contained in the source directories should be compiled in Bloop. This is a problem caused by Zinc as it can be reproduced in both sbt and bloop, though sbt does indeed recompile on file watch because it watches on all the files below the project base directory. This is one of the reasons why file watching in sbt is slow too.

**Comment by jvican** [comment]
Actually, no. sbt adds `x.scala` to the "sources" of the project and the file watcher gets to initialize the watcher with `x.scala`, but internally it ignores file paths and no file watching is performed. This is better than if the file paths wouldn't have ignored and would have been passed to the underlying JDK watch service because the watch service doesn't work for single files, only directories.
  
So, the only solution here is to watch the parent directory of `x.scala`. To make this operation faster, we can use the internal APIs of `directory-watcher` to only watch the directory in a non-recursive way.

**Comment by japgolly** [comment]
Yeah, it's not a bad little quirk/feature. Obviously you'd never use it in a proper project, but it's quite nice and convenient for providing runnable code snippets to compliment a blog or a presentation which was my use case.

**Comment by jvican** [comment]
I found a way to fix this @japgolly, so it will make it into 1.1.0. :smile:

**Comment by jvican** [comment]
This was fixed in #623.

**Comment by japgolly** [comment]
Very cool @jvican, thanks & great work as usual!
